Core/include/Acts/Geometry/LayerBlueprintNode.hpp

```diff
@@ -0,0 +1,141 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/StaticBlueprintNode.hpp"
+
+#include <ostream>
+
+namespace Acts {
+
+/// The layer node is essentially an auto-sizing wrapper around a set of
+/// surfaces.
+/// @note This implementation is **preliminary** and will likely change
+///       in the future.
+/// It defers most of the functionality to @ref Acts::StaticBlueprintNode,
+/// after the initial volume creation is completed.
+///
+/// The layer volume is created to wrap around the surfaces registered with
+/// this node. The orientation of the resulting volume defaults to the identity
+/// matrix. If another orientation is desired, this can be set with the @ref
+/// Acts::LayerBlueprintNode::setTransform. See @ref Acts::ProtoLayer for
+/// details on the auto-sizing from surfaces.
+///
+class LayerBlueprintNode : public StaticBlueprintNode {
+ public:
+  /// Enum that lists out the supported layer types.
+  enum class LayerType {
+    /// A cylinder layer
+    Cylinder,
```

Comment:
```suggestion
    Cylinder = 1,
```

I think this is what we usually do? Not sure

Reply:
Can do if you want, but I don't use the numerical value anywhere, so it shouldn't matter what gets assigned.